platforms/software/dependency-management/src/main/java/org/gradle/internal/component/local/model/TransformedComponentFileArtifactIdentifier.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.component.local.model;
+
+import org.gradle.api.artifacts.component.ComponentArtifactIdentifier;
+import org.gradle.api.artifacts.component.ComponentIdentifier;
+import org.gradle.internal.DisplayName;
+
+import java.util.Objects;
+
+/**
+ * Identifies the transformed artifact of a component. The original file name is tracked in order to guarantee uniqueness,
+ * as artifact transformations may result in multiple artifacts with the same file name.
+ */
+public class TransformedComponentFileArtifactIdentifier implements ComponentArtifactIdentifier, DisplayName {
+    private final ComponentIdentifier componentId;
+    private final String fileName;
+    private final String originalFileName;
+
+    public TransformedComponentFileArtifactIdentifier(ComponentIdentifier componentId, String fileName, String originalFileName) {
+        this.componentId = componentId;
+        this.fileName = fileName;
+        this.originalFileName = originalFileName;
+    }
+
+    @Override
+    public ComponentIdentifier getComponentIdentifier() {
+        return componentId;
+    }
+
+    public String getFileName() {
+        return fileName;
+    }
+
+    public String getOriginalFileName() {
+        return originalFileName;
+    }
+
+    @Override
+    public String getDisplayName() {
+        return getOriginalFileName() + " -> " + getFileName() + " (" + getComponentIdentifier().getDisplayName() + ")";
```

Comment:
❓ Where would you see those display names? Who is consuming them right now?

Reply:
Build scans do look like they get these display names to some extent. We have no tests verifying the display names we are sending to build scans, but it does look like this bubbles up to build operations in `TransformStepBuildOperation#describeSubject`. 

Its difficult to tell where else this affects build scans. 

As for other places, these are used for toString for artifacts returned form a resolution. They also appear in error messages and dependency verification reports

Reply:
`TransformStepBuildOperation#describeSubject` is not problematic, since the result ends up in `ExecutePlannedTransformStepBuildOperationType.Details.subjectType` which is not used in scans.